File: classify-server/src/main/java/io/ermdev/alice/repository/UserRepository.java
package io.ermdev.alice.repository;

import io.ermdev.alice.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;

public interface UserRepository extends JpaRepository<User, Long> {

    @Query("from User where id=:userId")
    User findById(@Param("userId") Long userId);

    @Query("from User")
    List<User> findAll();
}

File: src/main/kotlin/io/ermdev/classify/data/repository/UserRepository.kt
package io.ermdev.classify.data.repository

import io.ermdev.classify.data.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import java.util.*

interface UserRepository : JpaRepository<User, Long> {

    @Query("from User where username=:username")
    fun findByUsername(@Param("username") username: String): Optional<User>
}